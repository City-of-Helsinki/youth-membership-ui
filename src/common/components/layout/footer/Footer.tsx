import React from 'react';
import { Koros } from 'hds-react';
import { useTranslation } from 'react-i18next';

import Link from '../../../reactRouterWithLanguageSupport/Link';
import ExternalLink from '../../externalLink/ExternalLink';
import HelsinkiLogo from '../../helsinkiLogo/HelsinkiLogo';
import styles from './Footer.module.css';

function Footer() {
  const { t } = useTranslation();
  const year = new Date().getFullYear();
  return (
    <div className={styles.wrapper}>
      <Koros className={styles.koros} />
      <footer className={styles.footer}>
        <div className={styles.content}>
          <HelsinkiLogo className={styles.logo} />
          <hr />
          <div className={styles.textContainer}>
            <span>
              &copy; {t('footer.copyright', { year })} &bull;{' '}
              {t('footer.reserveRights')} &bull;{' '}
              <a href={t('registry.descriptionLink')} className={styles.links}>
                {t('footer.privacy')}
              </a>{' '}
              |Â 
              <Link to="/accessibility" className={styles.links}>
                {t('footer.accessibility')}
              </Link>
            </span>
            <ExternalLink
              href={t('footer.feedbackLink')}
              className={styles.feedback}
            >
              {t('footer.feedback')}
            </ExternalLink>
          </div>
        </div>
      </footer>
    </div>
  );
}

export default Footer;
