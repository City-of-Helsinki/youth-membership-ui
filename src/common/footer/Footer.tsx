import React from 'react';
import { Koros } from 'hds-react';
import { Link } from 'react-router-dom';
import { ReactComponent as HelsinkiLogo } from '../svg/HelsinkiLogo.svg';
import styles from './Footer.module.css';

function Footer() {
  return (
    <React.Fragment>
      <Koros className={styles.koros} />
      <footer className={styles.footer}>
        <div className={styles.content}>
          <HelsinkiLogo className={styles.logo} />
          <hr />
          <div className={styles.textContainer}>
            <span>
              &copy; Copyright 2020 &bull; Kaikki oikeudet pidätetään &bull;{' '}
              <Link to="/#" className={styles.links}>
                Rekisteriseloste
              </Link>{' '}
              | 
              <Link to="/#" className={styles.links}>
                Saavutettavuusseloste
              </Link>{' '}
              | 
              <Link to="/#" className={styles.links}>
                Palvelunkäyttöehdot
              </Link>
            </span>
            <Link to="/#" className={styles.feedback}>
              Anna palautetta
            </Link>
          </div>
        </div>
      </footer>
    </React.Fragment>
  );
}

export default Footer;
