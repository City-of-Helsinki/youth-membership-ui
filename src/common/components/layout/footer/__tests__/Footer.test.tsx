import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Footer from '../Footer';

test('matches snapshot', () => {
  const wrapper = shallow(<Footer />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
