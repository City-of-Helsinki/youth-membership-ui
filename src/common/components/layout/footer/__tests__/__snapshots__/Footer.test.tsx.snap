// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="wrapper"
>
  <Koros
    className="koros"
  />
  <footer
    className="footer"
  >
    <div
      className="content"
    >
      <HelsinkiLogo
        className="logo"
      />
      <hr />
      <div
        className="textContainer"
      >
        <span>
          © 
          © Helsingin kaupunki
           •
           
          Kaikki oikeudet pidätetään
           •
           
          <a
            className="links"
            href="https://www.hel.fi/static/liitteet/kanslia/rekisteriselosteet/Kuva/Kuva-EU-Nuorisopalvelujen-jasenrekisteri.pdf"
          >
            Rekisteriseloste
          </a>
           
          | 
          <Link
            className="links"
            to="/accessibility"
          >
            Saavutettavuusseloste
          </Link>
        </span>
        <a
          className="feedback"
          href="https://www.hel.fi/kulttuurin-ja-vapaa-ajan-toimiala/fi/yhteystiedot-ja-palaute/Anna+palautetta/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Anna palautetta
        </a>
      </div>
    </div>
  </footer>
</div>
`;
